src/conn/conn_compact.c

```diff
@@ -602,6 +616,17 @@ __background_compact_server(void *arg)
         if (!running)
             continue;
 
+        /*
+         * Throttle background compaction if any of the following conditions is met:
+         * - The dirty trigger threshold has been reached as compaction may generate more dirty
+         * content.
```

Comment:
Can we add a comment about why we don't consider update threshold here? We only mark the page dirty but not adding any new updates.

Reply:
Sure, added in 3f3320e6e1da3084d3c10e3eb9c48f44cd1aca65.